utils/clients/bitflyer/client.go

```diff
@@ -0,0 +1,288 @@
+package bitflyer
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/brave-intl/bat-go/settlement"
+	"github.com/brave-intl/bat-go/utils/altcurrency"
+	"github.com/brave-intl/bat-go/utils/clients"
+	"github.com/brave-intl/bat-go/utils/requestutils"
+	"github.com/shengdoushi/base58"
+	"github.com/shopspring/decimal"
+)
+
+// Quote returns a quote of BAT prices
+type Quote struct {
+	ProductCode  string          `json:"product_code"`
+	MainCurrency string          `json:"main_currency"`
+	SubCurrency  string          `json:"sub_currency"`
+	Rate         decimal.Decimal `json:"rate"`
+	PriceToken   string          `json:"price_token"`
+}
+
+// QuoteQuery holds the query params for the quote
+type QuoteQuery struct {
+	ProductCode string `url:"product_code,omitempty"`
+}
+
+// WithdrawToDepositIDPayload holds a single withdrawal request
+type WithdrawToDepositIDPayload struct {
+	CurrencyCode string  `json:"currency_code"`
+	Amount       float64 `json:"amount"`
+	DryRun       *bool   `json:"dry_run,omitempty"`
+	DepositID    string  `json:"deposit_id"`
+	TransferID   string  `json:"transfer_id"`
+	SourceFrom   string  `json:"source_from"`
+}
+
+// WithdrawToDepositIDBulkPayload holds all WithdrawToDepositIDPayload(s) for a single bulk request
+type WithdrawToDepositIDBulkPayload struct {
+	DryRun       bool                         `json:"dry_run"`
+	Withdrawals  []WithdrawToDepositIDPayload `json:"withdrawals"`
+	PriceToken   string                       `json:"price_token"`
+	DryRunOption *DryRunOption                `json:"dry_run_option"`
+}
+
+// WithdrawToDepositIDResponse holds a single withdrawal request
+type WithdrawToDepositIDResponse struct {
+	CurrencyCode string          `json:"currency_code"`
+	Amount       decimal.Decimal `json:"amount"`
+	Message      string          `json:"message"`
+	Status       string          `json:"transfer_status"`
+	TransferID   string          `json:"transfer_id"`
+}
+
+// TokenPayload holds the data needed to get a new token
+type TokenPayload struct {
+	GrantType         string `json:"grant_type"`
+	ClientID          string `json:"client_id"`
+	ClientSecret      string `json:"client_secret"`
+	ExtraClientSecret string `json:"extra_client_secret"`
+}
+
+// TokenResponse holds the response from refreshing a token
+type TokenResponse struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	ExpiresIn    int    `json:"expires_in"`
+	Scope        string `json:"scope"`
+	AccountHash  string `json:"account_hash"`
+	TokenType    string `json:"token_type"`
+}
+
+// DryRunOption holds options for dry running a transaction
+type DryRunOption struct {
+	RequestAPITransferStatus string `json:"request_api_transfer_status"`
+	ProcessTimeSec           uint   `json:"process_time_sec"`
```

Comment:
I would make this a duration instead of a uint.  If you did this the "ParseDryRun" function could be much reduced as Viper has a "GetDuration" method.  It seems that we are also overloading the dry run flag, sometimes it is boolean, sometimes it is a string representation of a number of seconds.  I think it would make more sense to have two flags, which are independent of each other.

```bash
--dry-run true --process-time 1s
```

Then we can ParseBool the dry-run and ParseDuration the process-time.

Reply:
fixed in 057e191